tests/functional/test_plugins.py

```diff
@@ -498,6 +497,146 @@ def test_metadata_azure(
     )
 
 
+@pytest.mark.parametrize("copy_files", [[LS_PATH]], indirect=True)
+@pytest.mark.parametrize("tmp_file", [{}], indirect=True)
+def test_metadata_gcp_no_vendor(
+    copy_files: list[Path],
+    chalk: Chalk,
+    tmp_file: Path,
+    server_imds: str,
+):
+    # make imds plugin think we are running in EC2
+    tmp_file.write_text("nothinghere")
+    bin_path = copy_files[0]
+    with open("/etc/resolv.conf", "a") as outf:
+        outf.write("search google.internal.\n")
```

Comment:
this can interfere with other tests as the resolv file is global

for easy testing might be worth adding a config in chalk where it should search for resolv file. this way we can customize it below similar to how we override `VENDOR` which tells chalk the path of the vendor file

Reply:
yeah but a search entry should be benign for the most part? having a separate config option just for this would require additional changes? or could it be just a user defined bit? If we are not make any other chalk changes and support it via a config as it is let's do it that way

Reply:
tests run in parallel so adjusting resolv can impact other tests running at the same time. previously we had issues with leaking tests so if we are touching anything global, we should handle that appropriately

Reply:
can we do this just w/ a custom chalk config without any other chalk changes?

Reply:
It is either there or not in most cases though so making it a param just for the sake of a test feels like an overkill. Most other resolv.conf equivalents are just updating /etc/resolv.conf usually https://tailscale.com/blog/sisyphean-dns-client-linux. Even in parallel tests I don't think an extra `search ` entry would impact anything?

Reply:
there are multiple concerns going on here:

* once we have multiple tests touching the same global dependency, they will all fight with each other. The worst part due to parallel tests it wont be deterministic which is why we had a bunch of issues like that before which were "fun" to figure out. Lets say azure implementation will also touch `resolv` file? Or well need multiple variants of GCP tests all running at the same time. They all will keep appending to the resolv file which wont be expected.
* currently we dont really support running tests directly from host but in eventually we should. Touching any host-level system things is not nice when running tests. Isolating it to a temporary file will ensure tests cannot possible impact system operation in anyway.

we already do that for the vendor file paths, even though their paths are also hardcoded per cloud provider but we use a config value to define the path allow:

* customize the path in case they run "interesting" setups like VMs
* easier testing

Reply:
exposed a `sys_resolv..` key under cloud_provider indentification, as it is used in this particular context